### DISCUSSION TITLE
How to generate an atrbitrary waveform python script with NiRFSG?

### DISCUSSION THREAD

**Discussion by markusmat97** [ni/grpc-device#821] [Q&A] [answered]
Hi
i found few examples here.https://github.com/ni/grpc-device/tree/main/examples/nirfsg.Would some can tell me if i want to generate an arbitrary waveform.How can i build a python script for this?
Do you have any guide to build python script?

**Comment by DavidCurtiss** [comment, marked as answer]
The examples here in the grpc-device repo are intended only to familiarize you with how to interact with the driver via gRPC. The entry points, though, largely mirror the C entry points, so it should generally be straightforward for you to adapt an example that you find elsewhere, like those mentioned [here](https://www.ni.com/docs/en-US/bundle/ni-rfsg/page/rfsg/installed_examples.html) or [here](https://www.ni.com/pdf/manuals/374769af.html#examples).

For example, after installing NI-RFSG, I found a C# example for arbitrary waveform generation here:
C:\Users\Public\Documents\National Instruments\NI-RFSG\Examples\VS2012\ArbitraryWaveforms\ArbitraryWaveformGeneration\cs\
In the MainForm.cs file there, there's a StartGeneration method that you could mimic.

It's usually going to be easiest to translate from the CVI example, though, if you install CVI support. I found the corresponding example here:
C:\Users\Public\Documents\National Instruments\CVI\samples\niRFSG\ArbitraryWaveforms\ArbitraryWaveformGeneration\

**Reply by markusmat97** [reply]
thank you for the reply.
i tried to find the CVI example but there was no CVI folder in my C:\Users\Public\Documents\National Instruments
How to get that?

**Reply by DavidCurtiss** [reply]
You have to check the box to install CVI support when you install (or modify ⚙️) NI-RFSG. But you might have to have CVI installed for that option to show up, I'm not actually sure.

**Comment by yardov** [comment]
Hi @markusmat97,

I just uploaded a script that shows this:
https://github.com/ni/grpc-device/blob/main/examples/nirfsg/rfsg-multitone-example.py
